### ISSUE TITLE
Show current screen brightness in Tweaking View

### ISSUE THREAD

**Issue by diotto** [edotassi/AmazMod#266]
When we open Tweaking View it shows the SeekBar and EditText for Brightness empty. It would be great if we could show the current watch brightness.

If not possible a second option would be to store in preferences the brightness when we set it using app and show it filled when we open Tweaking View.

**Comment by diotto** [comment]
Found this code in StackOverflow (https://stackoverflow.com/questions/33063718/getting-the-current-screen-brightness) :

```
Settings.System.putInt(
    cResolver,
    Settings.System.SCREEN_BRIGHTNESS_MODE,
    Settings.System.SCREEN_BRIGHTNESS_MODE_MANUAL);
brightness =
    Settings.System.getInt(
        cResolver,
        Settings.System.SCREEN_BRIGHTNESS);  
System.out.println("Current Brightness level " + brightness);
```

**Comment by edotassi** [comment]
could be added to the WatchStatus class

**Comment by diotto** [comment]
To check if automatic brightness is on (Source: https://stackoverflow.com/questions/6757728/how-to-get-the-status-of-automatic-brightness-is-on-or-off-in-settings-programma)

```
//if brigthnessmode=0 means Auto mode is currently off
//if brightnessmode=1 means Auto mode is currently on
try {
    brightnessmode = System.getInt(contentresolver, System.SCREEN_BRIGHTNESS_MODE);
} catch (Exception e) {
    Log.d("tag", e.toString());
}
```

**Comment by GreatApo** [comment]
To sum up: (these should work)

//ADB get values
adb shell settings get system screen_brightness_mode
adb shell settings get system screen_brightness
//ADB put values
adb shell settings put system screen_brightness_mode 1
adb shell settings put system screen_brightness 102

//JAVA get values
android.provider.Settings.System.getInt(getContentResolver(), "screen_brightness_mode");
android.provider.Settings.System.getInt(getContentResolver(), "screen_brightness");
//JAVA put values
android.provider.Settings.System.putInt(getContentResolver(), "screen_brightness_mode", 1);
android.provider.Settings.System.putInt(getContentResolver(), "screen_brightness", 100);

And the current value should be delivered from the "WatchStatus" data at the main screen.

**Comment by Saratoga79** [comment]
If
`brigthnessmode=0 means Auto mode is currently off`

Then, shouldn't it be 
`adb shell settings put system screen_brightness_mode 0`

In order to disable Auto Brightness and set the new value? Or am I understanding it wrong?

Ans same for java code?

**Comment by lfom** [comment]
@Saratoga79 

```
private static final String SCREEN_BRIGHTNESS_MODE = "screen_brightness_mode";
private static final int SCREEN_BRIGHTNESS_MODE_MANUAL = 0;
private static final int SCREEN_BRIGHTNESS_MODE_AUTOMATIC = 1;
```

**Comment by diotto** [comment]
Just commited to DEV the code to transport Brightness and BrightnessMode to phone. In Tweaking View it show the current brightness (if BrightnessMode is Auto it's showing the Seekbar end EditText disabled)

@edotassi and @lfom , i believe the code may need some refactoring, I used two constants in AmazmodApplication class to store the values, maybe there is a better way to store them. If possible take a look in my code and, if necessary, suggest corrections. Thanks!

**Comment by diotto** [comment]
People, is amazmod/com/transport/Constants.java used by service module ?

I have created the code bellow in both amazmod/com/transport/Constants.java and com/amazmod/service/Constants.java:

```
    public static final String SCREEN_BRIGHTNESS_MODE = "screen_brightness_mode";
    public static final int SCREEN_BRIGHTNESS_MODE_MANUAL = 0;
    public static final int SCREEN_BRIGHTNESS_MODE_AUTOMATIC = 1;
```

I believe it could be in only Constants.java inside amazmod/com/transport, but not sure.

**Comment by diotto** [comment]
Included a switch button for automatic brightness. 

Closed in commit [#8b2207e139c9e80814570f3a49a412cd3b0d89c8 ](https://github.com/edotassi/AmazMod/commit/8b2207e139c9e80814570f3a49a412cd3b0d89c8)
